hack/update_local_image.sh

```diff
@@ -7,14 +7,27 @@ set -o xtrace
 
 CLUSTER_CONTEXT=$(${KUBECTL} config current-context | cut -d'-' -f1)
 
+
+# This file is responsible to the update the local k8s (for example minikube) with the new image.
+# For minikube the faster option is to build the image after changing the docker environment to the minikube docker.
+# For k3d we can just import the image.
+# TODO Create a local docker registry push the image to it and remove this file
 case "${CLUSTER_CONTEXT}" in
   "minikube")
+    # In MACOS the minikube command cannot be mapped since it's a different executable.
+    if ! command -v minikube &> /dev/null
+    then
+      echo "minikube is not mapped (if you are using a MAC please ignore this message) - installing"
+      curl -LO https://storage.googleapis.com/minikube/releases/latest/minikube-latest.x86_64.rpm
```

Comment:
This is implicitly forcing an installation of a specific minikube version on the user machine, and also assuming the user is using a distro with an RPM package manager. I think it's best to leave it up to the user how they choose to install minikube on their machine, as [dictated by the README](https://github.com/openshift/assisted-service/blob/b5bdb320c77a509320a284e3b59033ea2656b8c8/README.md#L33)

Reply:
It's for MAC where you can't map the minikube to the container since it's a different executable like executing .exe file on a linux machine. Yuval changed the build image and removed the minikube executable and mapped it from the host machine.
It works on a linux hosts but not on a MAC

Reply:
![image](https://user-images.githubusercontent.com/10882062/115992736-e563b280-a5d7-11eb-9c26-b9101c3ab9dc.png)
It says that's it's not for MAC, is that a typo?

Reply:
no, If you are a MAC owner it's not an error so just ignore it

Reply:
I don't understand how that justifies implicitly installing RPMs on a user machine

Reply:
If we run our k8s on a minikube that installed on our host we must build the container on the minikube docker environment.
please see: https://minikube.sigs.k8s.io/docs/handbook/pushing/
In order to change our docker environment to the minikube docker we use the minikube docker-env command.
The minikube command by default (For a regular Fedora users) mapped to the skipper container by the skipper.yaml file.
(Just injecting the host executable to the container and eliminating the minikube installation on the container)
The only exception is for a MAC host where we can't run a macos executable on the linux container so we must install it
This code checks that we are using minikube as our k8s and ensures that minikube were installed.
Installing minikube is a expected behaviour even for a linux hosts (otherwise the docker build will succeed and the image would be created on the host registry instead of the minikube registry)

Reply:
Okay, I understand the reasoning better now.

This still assumes that the command/script will be ran via skipper. While skipper is convenient, it's not a required for running our scripts, if one chooses not to. So I don't really like that there's an RPM installation that is not part of the Dockerfile.

Please add a gate to make sure that you're running inside our skipper before installing that RPM.